smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/extension/AbstractEmitter.java

```diff
@@ -120,23 +128,27 @@ public boolean hasRequests() {
         int size = (int) defaultBufferSize;
         return stream
                 .onOverflow().buffer(size - 2)
-                .onFailure().invoke(t -> synchronousFailure.set(t));
+                .onFailure().invoke(synchronousFailure::set);
     }
 
     public Publisher<Message<? extends T>> getPublisher() {
         return publisher;
     }
 
-    boolean isSubscribed() {
-        return internal.get() != null;
-    }
-
     protected synchronized void emit(Message<? extends T> message) {
         if (message == null) {
             throw ex.illegalArgumentForNullValue();
         }
 
-        MultiEmitter<? super Message<? extends T>> emitter = verify(internal, name);
+        MultiEmitter<? super Message<? extends T>> emitter = verify();
+        if (emitter == null) {
+            if (overflow == OnOverflow.Strategy.DROP) {
+                // There are no subscribers, but because we use the DROP strategy, just ignore the event.
+                // However, nack the message, so the sender can be aware of the rejection.
+                message.nack(new IllegalStateException("Unable to process message - no subscriber"));
```

Comment:
isn't that costly? generating a new stacktrace on every message until there's a subscriber?

Reply:
Fixed. I used a constant exception that does not capture the stack trace.